testinfra/modules/sockethost.py

```diff
@@ -0,0 +1,120 @@
+# coding: utf-8
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import unicode_literals
+from __future__ import absolute_import
+
+import socket
+
+from testinfra.modules.base import Module
+
+
+def parse_socketspec(socketspec):
+    """Test listening tcp/udp sockets
+
+    ``socketspec`` must be specified as ``<protocol>://<host>:<port>``
+
+    Example:
+
+      - All ipv4 tcp sockets on port 22: ``tcp://22``
+      - All ipv4 sockets on port 22: ``tcp://0.0.0.0:22``
+      - udp socket on 127.0.0.1 port 69: ``udp://127.0.0.1:69``
+
+    """
+
+    protocol, address = socketspec.split("://", 1)
+
+    # if protocol not in ("udp", "tcp"):
+    #     raise RuntimeError(
+    #         "Cannot validate protocol '%s'. Should be tcp or udp" % (
+    #             protocol,))
+
+    if ":" in address:
+        # tcp://127.0.0.1:22
+        host, port = address.rsplit(":", 1)
+    else:
+        # tcp://22
+        host = None
+        port = address
+
+    if host is not None:
+        pass
+    else:
+        raise RuntimeError("Hostname is missing '%s'" % (host,))
+
+    if port is not None:
+        try:
+            port = int(port)
+        except ValueError:
+            raise RuntimeError("Cannot validate port '%s'" % (port,))
+
+    return protocol, host, port
+
+
+class SocketHost(Module):
```

Comment:
I'm not sure it's worth to add a new module for this. Maybe adding ``is_resolvable()`` and ``is_reachable()`` as classmethod to the existing ``Socket`` module could be better